internal/pkg/config/pbkdf2.go

```diff
@@ -0,0 +1,42 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package config
+
+import (
+	"bytes"
+	"errors"
+)
+
+type PBKDF2 struct {
+	Iterations int `config:"iterations"`
+	KeyLength  int `config:"key_length"`
+	SaltLength int `config:"salt_length"`
+
+	// BlockSize must be a factor of aes.BlockSize
+	BlockSize int `config:"block_size"`
+}
+
+// Validate the config options
+func (p *PBKDF2) Validate() error {
+	if p.Iterations == 0 {
+		return errors.New("iterations must be superior to 0")
+	}
+	if p.KeyLength == 0 {
+		return errors.New("key_length must be superior to 0")
+	}
+	if p.SaltLength == 0 {
+		return errors.New("salt_length must be superior to 0")
+	}
+	return nil
+}
+
+// InitDefaults is the default options to use with PDKDF2, changing might decrease
+// the efficacy of the encryption.
+func (p *PBKDF2) InitDefaults() {
+	p.Iterations = 10000
```

Comment:
fact: In 2023, OWASP recommended to use [210,000](https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html#pbkdf2) for PBKDF2-HMAC-SHA512. The more iterations the more time it will take to compute the hash, but not quite sure if we should follow this advise or not 🙂

Reply:
@pkoutsovasilis I was just using the defaults that the Elastic Agent uses so they can be inline. Should we also update the Elastic Agent.

https://github.com/elastic/elastic-agent/blob/main/internal/pkg/crypto/io.go#L58

Reply:
yy I ve seen that, however, I am not entirely sure how easy of a change is for the elastic-agent without breaking anything. Need to look into that 🙂

Reply:
Updated here at least.